### DISCUSSION TITLE
Trying to add templates for custom entity

### DISCUSSION THREAD

**Discussion by AliciaLevasseur** [sulu/sulu#7062] [Q&A] [answered]
Hi,
I'm trying to add a templates in a custom entity. Custom entity have default form détails but it possible to add more forms like template in Sulu Page ? I did not find any mention of a way to do that in the sulu documentation. Is there a way to do that, maybe a hook or something like that ?

**Comment by alexander-schranz** [comment]
Hello @AliciaLevasseur,

can you specify your usecase more. What kind of things are you trying to build? Do you want to extend a Custom Entity with whole Content Features (includes Templates, Drafting/Publishing, Multilanguage, ...).
What kind of requirements does your custom entity have?

**Comment by AliciaLevasseur** [comment]
Hello @alexander-schranz ,

I need an entity to store blogs articles like suluBundleArticle (but it is not compatible with my project : https://github.com/handcraftedinthealps/ElasticsearchBundle/pull/32). 

but waiting, i need an entity article with type, and for this I would like to use several models. My custom entity(without extend) is link to form for view/create/update details and is multilingual.
Is it possible to add more forms or templates for one entity ? 

My usecase is very similar to articleBundle, is it possible with custom entity to have similar result ?

**Reply by alexander-schranz** [reply, marked as answer]
You can normally easy downgrade the doctrine/annotations package by and install SuluArticleBundle via:

```bash
composer require sulu/article-bundle --no-update
composer update
```

Alternative to the SuluArticleBundle is a community project here by @TheCadien:

https://github.com/TheCadien/SuluNewsBundle

Adding template behaviour to any `FormView` in Sulu is done via an own structure configuration:

https://github.com/sulu/SuluContentBundle/blob/b5c3eeb2e6875437b81febcd8583e9c1400aacaf/Tests/Application/ExampleTestBundle/DependencyInjection/ExampleTestExtension.php#L33-L41

And a type switcher toolbar action: https://github.com/sulu/SuluContentBundle/blob/b5c3eeb2e6875437b81febcd8583e9c1400aacaf/Content/Infrastructure/Sulu/Admin/ContentViewBuilderFactory.php#L102

But to recreate whole functionality what pages provide more is required. We currently are working on an easy way to add content behaviour to any doctrine entity via the SuluContentBundle, whole example bundle can be found in it:

https://github.com/sulu/SuluContentBundle/tree/0.x/Tests/Application/ExampleTestBundle

But would recommend to stick currently with Article or News Bundle.
